activestorage/app/javascript/activestorage/blob_record.js

```diff
@@ -3,10 +3,11 @@ import { getMetaValue } from "./helpers"
 export class BlobRecord {
   constructor(file, checksum, url) {
     this.file = file
+    const content_type = file.type.length ? file.type : "application/octet-stream"
 
     this.attributes = {
       filename: file.name,
-      content_type: file.type,
+      content_type: content_type,
```

Comment:
Let's shorten and inline the default: `content_type: file.type || "application/octet-stream",`